legacy/fw_signatures.c

```
@@ -149,6 +150,21 @@ bool firmware_present_new(void) {
   return true;
 }
 
+bool firmware_present_upgrade(void) {
+  const image_header *hdr =
```

Clang (14.0.6)

[warning] variable 'hdr' is not initialized

(cppcoreguidelines-init-variables)